### PR TITLE
Use shader attributes for solid polygon layer

### PR DESCRIPTION
**Pull request by tsherif**
#### Background
Use `shaderAttributes` to replace SolidPolygonLayer attribute hijacking in `updateAttributes`. 

#### Change List
- Use new `shaderAttributes` instead of creating new attributes in `updateAttributes`. The `updateAttributes` override is is still necessary to switch to instance drawing for the sides.
- Update vertex shader module for SolidPolygonLayer to inject source code instead of branching. Was necessary to prevent the topModel from trying to access attributes it doesn't use.

### CODE REVIEW

**Comment by tsherif** [comment]
@Pessimistress much cleaner! Thanks for the `#define` tip :)

File: modules/layers/src/solid-polygon-layer/solid-polygon-layer.js [hunk 1]
```diff
@@ -228,20 +255,6 @@ export default class SolidPolygonLayer extends Layer {
           });
         }
       }
-      if (newAttributes.positions) {
```
Comment (Pessimistress) [inline modules/layers/src/solid-polygon-layer/solid-polygon-layer.js]: 🎉

File: modules/layers/src/solid-polygon-layer/solid-polygon-layer.js [hunk 2]
```diff
@@ -87,9 +94,29 @@ export default class SolidPolygonLayer extends Layer {
         transition: ATTRIBUTE_TRANSITION,
         accessor: 'getPolygon',
         update: this.calculatePositions,
-        noAlloc
+        shaderAttributes: {
+          positions: {
+            size: 3
```
Comment (Pessimistress) [inline modules/layers/src/solid-polygon-layer/solid-polygon-layer.js]: Is the `size` override necessary?

Comment (tsherif) [reply modules/layers/src/solid-polygon-layer/solid-polygon-layer.js]: It's not, though having a 'positions' key is necessary. I figure this is a bit clearer than having an empty object?

**Pull request merged**